core/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -323,6 +323,39 @@ public Settings onNodeStopped(String nodeName) throws Exception {
     }
 
     /**
+     * This test ensures that when an index deletion takes place while a node is offline, when that
+     * node rejoins the cluster, it deletes the index locally instead of importing it as a dangling index.
+     */
+    public void testIndexDeletionWhenNodeRejoins() throws Exception {
+        final String indexName = "test-index-del-on-node-rejoin-idx";
+        logger.info("--> starting two nodes");
+        internalCluster().startNodesAsync(2).get();
+
+        logger.info("--> create an index");
+        createIndex(indexName);
+
+        logger.info("--> waiting for green status");
+        ensureGreen();
+
+        logger.info("--> stop the non-master node");
+        final String stoppedNode = internalCluster().removeRandomNonMasterFromCluster();
```

Comment:
I think we want to stop a random data node, right? Why do we care about a master?